plugin-k8s/src/lib.rs

```diff
@@ -84,69 +104,153 @@ impl AlumetPlugin for K8sPlugin {
                             let mut path_cpu = path.clone();
+                            let full_name_to_seek = pod_uid.strip_suffix(".slice").unwrap_or(&pod_uid);
+                            let parts: Vec<&str> = full_name_to_seek.split("pod").collect();
+                            let name_to_seek_raw = *(parts.last().unwrap_or(&full_name_to_seek));
+                            let uid_raw = parts.last().unwrap_or(&"No UID found");
+                            let uid = format!("pod{}", uid_raw);
+                            let name_to_seek = name_to_seek_raw.replace("_", "-");
+                            // let (name, ns) = cgroup_v2::get_pod_name(name_to_seek.to_owned());
+                            let hostname_ostring = gethostname();
+                            let hostname = hostname_ostring
+                                .to_str()
+                                .context("Invalid UTF-8 in Hostname")
+                                .unwrap_or("")
+                                .to_string();
+                            let rt = match tokio::runtime::Builder::new_current_thread().enable_all().build() {
+                                Ok(rt_ok) => rt_ok,
+                                Err(err) => {
+                                    log::error!("Runtime failed to build and returned an error: {}", err);
+                                    return;
+                                }
+                            };
+                            let (name, namespace, node) = match rt.block_on(async {
+                                cgroup_v2::get_pod_name(
+                                    name_to_seek.to_owned(),
+                                    hostname,
+                                    self.kubernetes_api_url.clone(),
+                                )
+                                .await
+                            }) {
+                                Ok(tuple_found) => tuple_found,
+                                Err(err) => {
+                                    log::error!("Block on failed returned an error: {}", err);
+                                    return;
+                                }
+                            };
+
                             path_cpu.push("cpu.stat");
-                            let file = File::open(&path_cpu).with_context(|| format!("failed to open file {}", path_cpu.display())).unwrap();
+                            let file = match File::open(&path_cpu)
+                                .with_context(|| format!("failed to open file {}", path_cpu.display()))
+                            {
+                                Ok(file_opened) => file_opened,
+                                Err(err) => {
+                                    log::error!("Failed to open file at {:?}, it returned an error: {}", path_cpu, err);
+                                    return;
+                                }
+                            };
```

Comment:
`?` instead of a manual return would be better (that requires you to create another function, but it's fine, you can make it local to handle_event)